client/blocks/login/two-factor-authentication/security-key-form.jsx

```diff
@@ -0,0 +1,103 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormButton from 'components/forms/form-button';
+import { localize } from 'i18n-calypso';
+import { recordTracksEventWithClientId as recordTracksEvent } from 'state/analytics/actions';
+import { formUpdate, loginUserWithSecurityKey } from 'state/login/actions';
+import TwoFactorActions from './two-factor-actions';
+import Spinner from 'components/spinner';
+
+/**
+ * Style dependencies
+ */
+import './verification-code-form.scss';
+
+class SecurityKeyForm extends Component {
+	static propTypes = {
+		formUpdate: PropTypes.func.isRequired,
+		loginUserWithSecurityKey: PropTypes.func.isRequired,
+		onSuccess: PropTypes.func.isRequired,
+		recordTracksEvent: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	state = {
+		isDisabled: false,
+	};
+
+	initiateSecurityKeyAuthentication = event => {
+		event.preventDefault();
+
+		const { onSuccess } = this.props;
+		this.setState( { isDisabled: true } );
+		this.props
+			.loginUserWithSecurityKey()
+			.then( () => onSuccess() )
+			.catch( () => this.setState( { isDisabled: false } ) );
```

Comment:
Yes, I think it's an intended side effect of `onSuccess` (currently it results on a browser redirect). This was recently changed in c6a4b31.

Reply:
Yup when we used `finally()` we got a flash of the spinner going away to the click button to start key auth card before getting redirected to logged in view which felt weird / broken so I changed it to only hide the waiting spinner if an auth failure happened so that the user can retry with another key.